sympy/core/equation.py

```diff
@@ -0,0 +1,342 @@
+        return self._binary_op(self, other, lambda a, b: a * b)
+
+    def __rmul__(self, other):
+        return self._binary_op(other, self, lambda a, b: a * b)
+
+    def __sub__(self, other):
+        return self._binary_op(self, other, lambda a, b: a - b)
+
+    def __rsub__(self, other):
+        return self._binary_op(other, self, lambda a, b: a - b)
+
+    def __truediv__(self, other):
+        return self._binary_op(self, other, lambda a, b: a / b)
+
+    def __rtruediv__(self, other):
+        return self._binary_op(other, self, lambda a, b: a / b)
+
+    def __mod__(self, other):
+        return self._binary_op(self, other, lambda a, b: a % b)
+
+    def __rmod__(self, other):
+        return self._binary_op(other, self, lambda a, b: a % b)
+
+    def __pow__(self, other):
+        return self._binary_op(self, other, lambda a, b: a ** b)
+
+    def __rpow__(self, other):
+        return self._binary_op(other, self, lambda a, b: a ** b)
+
+    def _eval_power(self, other):
+        return self.__pow__(other)
+
+    #####
+    # Output helper functions
+    #####
+    def __repr__(self):
+        return str(self.lhs) + '=' + str(self.rhs)
+
+    def _latex(self, obj, **kwargs):
+        from sympy.printing import latex
+        return latex(self.lhs, **kwargs) + '=' + latex(self.rhs, **kwargs)
+
+    def __str__(self):
+        return self.__repr__()
+
+    #####
+    # Operation helper functions
+    #####
+    def expand(self, *args, **kwargs):
+        return Equation(self.lhs.expand(*args, **kwargs), self.rhs.expand(
+            *args, **kwargs), check=False)
+
+    def simplify(self, *args, **kwargs):
+        return self._eval_simplify(*args, **kwargs)
+
+    def _eval_simplify(self, *args, **kwargs):
+        return Equation(self.lhs.simplify(*args, **kwargs), self.rhs.simplify(
+            *args, **kwargs), check=False)
+
+    def _eval_factor(self, *args, **kwargs):
+        # TODO: cancel out factors common to both sides.
+        return Equation(self.lhs.factor(*args, **kwargs), self.rhs.factor(
+            *args, **kwargs), check=False)
+
+    def factor(self, *args, **kwargs):
+        return self._eval_factor(*args, **kwargs)
+
+    def _eval_collect(self, *args, **kwargs):
+        from sympy.simplify.radsimp import collect
+        return Equation(collect(self.lhs, *args, **kwargs),
+                        collect(self.rhs, *args, **kwargs), check=False)
+
+    def collect(self, *args, **kwargs):
+        return self._eval_collect(*args, **kwargs)
+
+    def evalf(self, *args, **kwargs):
+        return Equation(self.lhs.evalf(*args, **kwargs),
+                        self.rhs.evalf(*args, **kwargs), check=False)
+
+    def _eval_derivative(self, *args, **kwargs):
+        # TODO Find why diff and Derivative do not appear to pass through
+        #  kwargs to this. Since we cannot set evaluation of lhs manually
+        #  try to be intelligent about when to do it.
+        from sympy.core.function import Derivative
```

Comment:
`Derivative` and `Integrals` objects do not store the keyword arguments. Once the constructor complete its work, those kwargs are lost! More so, if you look in the core module `function.py`, inside the class `Derivative` where the method `_eval_derivative` is called, no `kwargs` are passed in! With this in mind, I believe it's impossible to actually chose which side to evaluate by using the methods `_eval_derivative` and `_eval_integral`. At the moment, the only solution I see is to override the methods `diff` and `integrate`: inside them, create `Derivative` or `Integral` for both sides, then perform the evaluation based on the value of `side`.

Reply:
I believe you are correct. I am leaving things as are for now.